src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AssemblyAlignmentParser.java

```diff
@@ -0,0 +1,174 @@
+
+        return ctx.textFile(pathToInputAlignments).map(textLine -> AlignmentRegion.fromString(textLine.split(AlignmentRegion.STRING_REP_SEPARATOR,-1)))
+                .flatMap(oneRegion -> breakGappedAlignment(oneRegion, SVConstants.GAPPED_ALIGNMENT_BREAK_DEFAULT_SENSITIVITY))
+                .mapToPair(alignmentRegion -> new Tuple2<>(new Tuple2<>(alignmentRegion.assemblyId, alignmentRegion.contigId), alignmentRegion))
+                .groupByKey();
+    }
+
+    // TODO: 12/15/16 test
+    static JavaPairRDD<Tuple2<String, String>, byte[]> loadContigSequence(final JavaSparkContext ctx,
+                                                                          final String pathToInputAssemblies) {
+        return loadContigsCollectionKeyedByAssemblyId(ctx, pathToInputAssemblies)
+                .flatMapToPair(assemblyIdAndContigsCollection -> {
+                    final String assemblyId = assemblyIdAndContigsCollection._1;
+                    final ContigsCollection contigsCollection = assemblyIdAndContigsCollection._2;
+                    return contigsCollection.getContents().stream()
+                            .map(pair -> new Tuple2<>(new Tuple2<>(assemblyId, pair._1.toString()), pair._2.toString().getBytes()))
+                            .collect(Collectors.toList());
+                });
+    }
+
+    // TODO: 11/26/16 assuming 'I'/'D' cannot be the beginning or last CIGAR operator AND the following operator cannot be a clipping
+    /**
+     * Break a gapped alignment into multiple alignment regions, based on input sensitivity: i.e. if the gap(s) in the input alignment
+     * is equal to or larger than the size specified, two alignment regions will be generated.
+     * @return an iterable of size >= 1. if size==1, the returned iterable contains only the input (i.e. either no gap or hasn't reached sensitivity)
+     */
+    @VisibleForTesting
+    static Iterable<AlignmentRegion> breakGappedAlignment(final AlignmentRegion oneRegion, final int sensitivity) {
+
+        final List<CigarElement> cigarElements = oneRegion.forwardStrandCigar.getCigarElements();
+        if(cigarElements.size()==1) return new ArrayList<>( Collections.singletonList(oneRegion) );
+        final int gapCount = (int) cigarElements.stream().filter(ce -> ce.getOperator().isIndel() && ce.getLength() >= sensitivity).count();
+        if (gapCount==0) return new ArrayList<>( Collections.singletonList(oneRegion) );
+
+        // get necessary information for computing
+        final String asmId = oneRegion.assemblyId;
+        final String contigId = oneRegion.contigId;
+        final String chr = oneRegion.referenceInterval.getContig();
+        final Cigar originalForwardStrandCigar = oneRegion.forwardStrandCigar;
+        final boolean isForwardStrand = oneRegion.forwardStrand;
+        final int newMapQual = oneRegion.mapQual;
```

Comment:
It was my undecidedness on what to do with mapping qualities when breaking the alignment. Technically I think (might be wrong) the mapping quality, since some `M` part of the original alignment is removed, is possibly affected, the mapping quality is not necessarily going to be the same as when the `M` part was there.